six/modules/c++/scene/source/ProjectionPolynomialFitter.cpp

```diff
@@ -102,6 +103,155 @@ ProjectionPolynomialFitter::ProjectionPolynomialFitter(
     }
 }
 
+ProjectionPolynomialFitter::ProjectionPolynomialFitter(
+        const ProjectionModel& projModel,
+        const GridECEFTransform& gridTransform,
+        const types::RowCol<size_t>& fullExtent,
+        const types::RowCol<double>& outPixelStart,
+        const types::RowCol<size_t>& outExtent,
+        const std::vector<types::RowCol<double> >& polygon,
+        size_t numPoints1D) :
+    mNumPoints1D(numPoints1D),
+    mOutputPlaneRows(numPoints1D, numPoints1D),
+    mOutputPlaneCols(numPoints1D, numPoints1D),
+    mSceneCoordinates(numPoints1D,
+                      numPoints1D,
+                      types::RowCol<double>(0.0, 0.0)),
+    mTimeCOA(numPoints1D, numPoints1D)
+{
+    // Get bounding rectangle of output plane polygon.
+    double minRow =  std::numeric_limits<double>::max();
+    double maxRow = -std::numeric_limits<double>::max();
+    double minCol =  std::numeric_limits<double>::max();
+    double maxCol = -std::numeric_limits<double>::max();
+
+    for (size_t ii = 0; ii < polygon.size(); ++ii)
+    {
+        minRow = std::min(minRow, polygon[ii].row);
+        maxRow = std::max(maxRow, polygon[ii].row);
+        minCol = std::min(minCol, polygon[ii].col);
+        maxCol = std::max(maxCol, polygon[ii].col);
+    }
+
+    if (minRow > static_cast<double>(fullExtent.row) ||
+        maxRow < 0 ||
+        minCol > static_cast<double>(fullExtent.col) ||
+        maxCol < 0)
+    {
+        throw except::Exception(Ctxt(
+            "Bounding rectangle is outside of output extent"));
+    }
+
+    // Only interested in pixels inside the fullExtent.
+    minRow = std::max(minRow, 0.0);
+    minCol = std::max(minCol, 0.0);
+    maxRow = std::min(maxRow, static_cast<double>(fullExtent.row) - 1);
+    maxCol = std::min(maxCol, static_cast<double>(fullExtent.col) - 1);
+
+    // Get size_t extent of the set of points.
+    const size_t minRowI = static_cast<size_t>(std::ceil(minRow));
+    const size_t minColI = static_cast<size_t>(std::ceil(minCol));
+    const size_t maxRowI = static_cast<size_t>(std::floor(maxRow));
+    const size_t maxColI = static_cast<size_t>(std::floor(maxCol));
+
+    if (minRowI > maxRowI || minColI > maxColI)
+    {
+        throw except::Exception(Ctxt(
+            "Bounding rectangle has no area"));
+    }
+
+    // The offset and extent are relative to the entire global output plane.
+    types::RowCol<double> boundingOffset(
+        static_cast<double>(minRowI),
+        static_cast<double>(minColI));
+    types::RowCol<size_t> boundingExtent(maxRowI - minRowI + 1,
+                                         maxColI - minColI + 1);
+
+    // No offset for the polygon mask.
+    types::RowCol<sys::SSize_T> pmOffset(0,0);
```

Comment:
Just remove this variable and let the PolygonMask constructor default that third argument